src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -449,6 +448,9 @@ public ModuleHandle GetModuleHandle()
         [MethodImpl(MethodImplOptions.InternalCall)]
         internal static extern bool CompareCanonicalHandles(RuntimeType left, RuntimeType right);
 
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        internal static extern object? GetThreadLocalAllocInternal(MethodTable* pMT, out uint size);
```

Comment:
```suggestion
        internal static extern object? InternalAllocNoChecks_FastPath(MethodTable* pMT);
```
It is not really important that the allocation is done from thread-local quota if possible. I would use a more generic fast path / slow path naming, similar to one of the similar existing splits.